src/Button/Button.tsx

```diff
@@ -126,33 +89,63 @@ export const Button: FC<ButtonProps> = forwardRef<
         >
           {children}
         </LegacyButton>
-      )}
-    </div>
-  ),
+      )
+    }
+
+    return (
+      <Container
+        id={id}
+        className={className}
+        disabled={disabled || loading}
+        onClick={(e) => {
+          handleClick && handleClick(e)
+        }}
+        isLoading={loading}
+        primary={primary}
+        secondary={secondary}
+        tertiary={tertiary}
+        icon={icon}
+        forcedWidth={forcedWidth}
+        {...(form ? { form } : {})}
+        type={type}
+        {...props}
+        ref={ref}
+      >
+        {loading && (
+          <LoaderContainer>
+            <Loader
+              color={primary ? 'white' : `${theme.colors.secondary}`}
+              height="16"
+            />
+          </LoaderContainer>
+        )}
+        <ContentContainer icon={icon} loading={loading}>
+          {icon && (
+            <IconContainer
+              render={icon}
+              size={24}
+              color={primary ? 'white' : 'secondary'}
+            />
+          )}
+          <ChildrenContainer>{children}</ChildrenContainer>
+        </ContentContainer>
```

Comment:
The content is now never removed to avoid the size to change during state change, instead we make the content transparent when in a loading state